subworkflows/local/prepare_genome.nf

```diff
@@ -44,78 +44,95 @@ workflow BUILD_INDICES {
+        result_fai = SAMTOOLS_FAIDX.out.fai
+        ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)
+    }
 
-    result_target_bed  = Channel.empty()
-    version_target_bed = Channel.empty()
+    ch_tabix_versions = Channel.empty()
+
+    result_target_bed = Channel.empty()
     if ((params.target_bed) && ('manta' in tools || 'strelka' in tools)) {
         target_bed_id = target_bed.map{ it -> [[id:it[0].getName()], it] }
-        (result_target_bed, version_target_bed) = TABIX_BGZIPTABIX(target_bed_id)
-        result_target_bed = result_target_bed.map{ meta, bed, tbi -> [bed, tbi] }
+        TABIX_BGZIPTABIX(target_bed_id)
+        result_target_bed = TABIX_BGZIPTABIX.out.tbi.map{ meta, bed, tbi -> [bed, tbi] }
+        ch_tabix_versions = ch_tabix_versions.mix(TABIX_BGZIPTABIX.out.versions).first()
     }
 
-    result_dbsnp_tbi  = Channel.empty()
-    version_dbsnp_tbi = Channel.empty()
+    result_dbsnp_tbi = Channel.empty()
     if (!(params.dbsnp_tbi) && params.dbsnp && ('mapping' in step || 'preparerecalibration' in step || 'controlfreec' in tools || 'haplotypecaller' in tools|| 'mutect2' in tools || 'tnscope' in tools)) {
         dbsnp_id = dbsnp.map{ it -> [[id:it[0].baseName], it] }
-        (result_dbsnp_tbi, version_dbsnp_tbi) = TABIX_DBSNP(dbsnp_id)
-        result_dbsnp_tbi = result_dbsnp_tbi.map{ meta, tbi -> [tbi] }
+        TABIX_DBSNP(dbsnp_id)
+        result_dbsnp_tbi = TABIX_DBSNP.out.tbi.map{ meta, tbi -> [dbsnp, tbi] }
+        ch_tabix_versions = ch_tabix_versions.mix(TABIX_DBSNP.out.versions).first()
     }
 
-    result_germline_resource_tbi  = Channel.empty()
-    version_germline_resource_tbi = Channel.empty()
+    result_germline_resource_tbi = Channel.empty()
     if (!(params.germline_resource_tbi) && params.germline_resource && 'mutect2' in tools) {
         germline_resource_id = germline_resource.map{ it -> [[id:it[0].baseName], it] }
-        (result_germline_resource_tbi, version_germline_resource_tbi) = TABIX_GERMLINE_RESOURCE(germline_resource_id)
+        TABIX_GERMLINE_RESOURCE(germline_resource_id)
+        result_germline_resource_tbi = TABIX_GERMLINE_RESOURCE.out.tbi.map{ meta, tbi -> [germline_resource, tbi] }
+        ch_tabix_versions = ch_tabix_versions.mix(TABIX_GERMLINE_RESOURCE.out.versions).first()
     }
 
-    result_known_indels_tbi  = Channel.empty()
-    version_known_indels_tbi = Channel.empty()
+    result_known_indels_tbi = Channel.empty()
     if (!(params.known_indels_tbi) && params.known_indels && ('mapping' in step || 'preparerecalibration' in step)) {
         known_indels_id = known_indels.map{ it -> [[id:it[0].baseName], it] }
-        (result_known_indels_tbi, version_known_indels_tbi) = TABIX_KNOWN_INDELS(known_indels_id)
-        result_known_indels_tbi = result_known_indels_tbi.map{ meta, tbi -> [tbi] }
+        TABIX_KNOWN_INDELS(known_indels_id)
+        result_known_indels_tbi = TABIX_KNOWN_INDELS.out.tbi.map{ meta, tbi -> [known_indels, tbi] }
```

Comment:
```suggestion
        result_known_indels_tbi.mix(TABIX_KNOWN_INDELS.out.tbi.map{ meta, tbi -> [known_indels, tbi] })
```
Does this work? `known_indels` should be a channel, not a data element.

Reply:
It seems it's implicitly turned into a channel only in the process call, not in `take:`.

Reply:
It is working, we tried that with @FriederikeHanssen
But we might have to do some adjustments later